sample-compose/src/main/java/com/airbnb/mvrx/compose/sample/ComposeSampleActivity.kt

```diff
@@ -0,0 +1,99 @@
+package com.airbnb.mvrx.compose.sample
+
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.material.Button
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.LifecycleOwner
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.airbnb.mvrx.MavericksState
+import com.airbnb.mvrx.MavericksViewModel
+import com.airbnb.mvrx.compose.collectState
+import com.airbnb.mvrx.compose.mavericksViewModel
+
+data class CounterState(
+    val count: Int = 0,
+) : MavericksState
+
+class CounterViewModel(initialState: CounterState) : MavericksViewModel<CounterState>(initialState) {
+    fun incrementCount() = setState { copy(count = count + 1) }
+}
+
+class ComposeSampleActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            Column {
+                Box(modifier = Modifier.weight(1f)) {
+                    CounterScreenNavHost("Counter Screen 1")
+                }
+                Divider()
+                Box(modifier = Modifier.weight(1f)) {
+                    CounterScreenNavHost("Counter Screen 2")
+                }
+            }
+        }
+    }
+
+    @Composable
+    fun CounterScreenNavHost(title: String) {
+        val navController = rememberNavController()
+        NavHost(navController, startDestination = "counter") {
+            composable("counter") {
+                CounterScreen(title)
+            }
+        }
+    }
+
+    @Composable
+    fun CounterScreen(title: String) {
+        // This will get or create a ViewModel scoped to the closest LocalLifecycleOwner which is the NavHost.
+        val navScopedViewModel: CounterViewModel = mavericksViewModel()
+        // This will get or create a ViewModel scoped to the Activity.
+        val activityScopedViewModel: CounterViewModel = mavericksViewModel(LocalContext.current as LifecycleOwner)
```

Comment:
the syntax `mavericksViewModel(LocalContext.current as LifecycleOwner)` for getting an activity scoped view model doesn't seem too intuitive to me

Reply:
Agreed. Do you have other ideas?

Reply:
I added `mavericksActivityViewModel()`

Reply:
I like `mavericksActivityViewModel`. Maybe also `mavericksLocalViewModel`? ( suppose that would replace fragment scoping).

i think this is something we can leave out for now though until we get a better handle on use cases